stl/inc/limits

```diff
@@ -1058,9 +1058,14 @@ _NODISCARD int _Checked_x86_x64_countr_zero(const _Ty _Val) noexcept {
     constexpr _Ty _Max    = (numeric_limits<_Ty>::max)();
 
 #ifndef __AVX2__
-    const bool _Definitely_have_tzcnt = __isa_available >= __ISA_AVAILABLE_AVX2;
-    if (!_Definitely_have_tzcnt && _Val == 0) {
-        return _Digits;
+    // Because the widening done below will always give a non-0 value, checking for tzcnt
+    // is not required on 16-bits and smaller since the only difference in behavior between

```

Comment:
Thanks for the excellent explanation! I have a couple of suggested improvements:

* "required for" sounds better for input types; "required on" would be more suitable for platforms such as x86/x64
* Instead of "and smaller", we can enumerate the whole list of widths

```suggestion
    // is not required for 8-bit and 16-bit since the only difference in behavior between
```

Reply:
My English isn't perfect, thanks for the corrections!